optuna/study/_tell.py

```diff
@@ -123,23 +123,6 @@ def _tell_with_warning(
             trial_id = study._storage.get_trial_id_from_study_id_trial_number(
                 study._study_id, trial_number
             )
-        except NotImplementedError as e:
-            warnings.warn(
-                "Study.tell may be slow because the trial was represented by its number but "
-                f"the storage {study._storage.__class__.__name__} does not implement the "
-                "method required to map numbers back. Please provide the trial object "
-                "to avoid performance degradation."
-            )
-
```

Comment:
Do you think it is okay to drop this error message? (In the current interface, keeping this warning seems to be a bit difficult.)

Reply:
Yes. Even in the current interface it is possible to emit a warning message in `BaseStorage.get_trial_id_from_study_id_trial_number()`, but I don't think we should do that.

BaseStorage already provides the default logic in some storage methods, which may cause a performance degradation (e.g. `get_trial_number_from_id`, `get_n_trials`, `get_best_trial`, etc.). However, they do not emit warning messages actually. `RDBStorage` can be made faster by overriding these methods, but if we use InMemoryStorage or CachedStorage, this default logic would not be a bottleneck.

Reply:
Thanks for elaborating. That makes sense. Let's drop this error message for consistency with the other default methods 👍